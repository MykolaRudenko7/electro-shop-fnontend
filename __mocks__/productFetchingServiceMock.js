export const productFetchingServiceMock = {
  productsSuccessResponse: {
    data: [
      {
        inStock: true,
        slug: 'mi-lite-15',
        url: 'https://res.cloudinary.com/dqdonqby4/image/upload/v1612476949/new-products/1_hw0qj0.jpg',
        name: 'Laptop Mi Notebook Lite 15 I5/8/512/MX110/W (JYU4139CN)',
        reviewsCount: 10,
        previousPrice: 29999,
        rating: 4.3,
        price: 21499,
      },
    ],
  },

  laptopsErrorResponse: {
    status: 400,
    statusText: 'Bad Request',
    data: { message: 'Laptops not found' },
  },

  newProductsErrorResponse: {
    status: 400,
    statusText: 'Bad Request',
    data: { message: 'New Products not found' },
  },
}
