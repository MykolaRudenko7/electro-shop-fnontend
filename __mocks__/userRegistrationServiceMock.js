const {
  TEST_EMAIL,
  TEST_PASSWORD,
  TEST_SIGN_UP_NAME,
  TEST_SIGN_UP_MOBILE_NUMBER,
  TEST_REFRESH_TOKEN,
  TEST_ACCESS_TOKEN,
  TEST_HASH_PASSWORD,
  TEST_USER_ID,
  TEST_ACTIVATION_LINK,
} = process.env

export const userRegistrationServiceMock = {
  userLoginResponse: {
    data: [
      {
        accessToken: TEST_ACCESS_TOKEN,
        message: 'Log in successfully',
        refreshToken: TEST_REFRESH_TOKEN,
        success: true,
        user: {
          activationlink: TEST_ACTIVATION_LINK,
          email: TEST_EMAIL,
          isActivated: false,
          lockUntil: null,
          loginAttempts: 0,
          mobileNumber: TEST_SIGN_UP_MOBILE_NUMBER,
          name: TEST_SIGN_UP_NAME,
          password: TEST_HASH_PASSWORD,
          __v: 0,
          _id: TEST_USER_ID,
        },
      },
    ],
    status: 200,
  },
  signUpUserRequest: {
    name_newUser: TEST_SIGN_UP_NAME,
    email_newUser: TEST_PASSWORD,
    password_newUser: TEST_PASSWORD,
    confirmPassword_newUser: TEST_PASSWORD,
    mobileNumber_newUser: TEST_SIGN_UP_MOBILE_NUMBER,
  },
  badSignUpUserRequest: {
    name_newUser: TEST_SIGN_UP_NAME,
    email_newUser: 'wrongEmail@gmail.com',
    password_newUser: 'wrongPassword',
    confirmPassword_newUser: 'wrongPassword',
    mobileNumber_newUser: '+wrongNumber',
  },
  signInUserRequest: {
    email_signIn: TEST_EMAIL,
    password_signIn: TEST_PASSWORD,
  },
  badSignInUserRequest: {
    email_signIn: TEST_EMAIL,
    password_signIn: 'wrongPassword',
  },
  logOutUserResponse: { success: true, message: 'Log out successfully' },

  errorSignInResponse: { status: 400, message: 'User with email does not exist' },
  errorSignUpResponse: { status: 400, message: 'User with this email or phone already exists' },
  errorLogOutResponse: { status: 400, message: 'Network Error' },
  unauthorizedErrorResponse: { status: 401, message: 'User is not authorized' },
}
