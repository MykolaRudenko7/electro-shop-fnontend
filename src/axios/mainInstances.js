import Axios from 'axios'
import { setupCache } from 'axios-cache-interceptor'

const { API_BASE_URL } = process.env

const mainApiClient = Axios.create({
  baseURL: API_BASE_URL,
})

export const axiosInstanceCache = setupCache(mainApiClient)
