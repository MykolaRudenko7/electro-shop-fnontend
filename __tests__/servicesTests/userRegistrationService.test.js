import axios from 'axios'
import UserRegistrationService from 'src/services/userRegistrationService'
import { userRegistrationServiceMock } from '__mocks__/userRegistrationServiceMock'

const {
  userLoginResponse,
  logOutUserResponse,
  signInUserRequest,
  signUpUserRequest,
  errorSignInResponse,
  errorSignUpResponse,
  errorLogOutResponse,
  unauthorizedErrorResponse,
  badSignInUserRequest,
  badSignUpUserRequest,
} = userRegistrationServiceMock

jest.mock('axios', () => {
  return {
    create: jest.fn(() => axios),
    post: jest.fn(() => Promise.resolve()),
    get: jest.fn(() => Promise.resolve()),
  }
})

describe('Auth service', () => {
  it('should return user data after successful sign in request', async () => {
    axios.post = jest.fn().mockResolvedValue(userLoginResponse)

    const userData = await UserRegistrationService.signInUser(signInUserRequest)

    expect(axios.post).toHaveBeenCalledTimes(1)
    expect(userData).toEqual(userLoginResponse)
    expect(userData).toMatchSnapshot()
  })

  it('should return an object with error status and message if sign-in request is incorrect', async () => {
    axios.post.mockImplementationOnce(() => Promise.reject(errorSignInResponse))

    await expect(UserRegistrationService.signInUser(badSignInUserRequest)).rejects.toMatchObject(
      errorSignInResponse,
    )
    expect(badSignInUserRequest).toMatchSnapshot()
  })

  it('should return user data after successful sign up request', async () => {
    axios.post = jest.fn().mockResolvedValue(userLoginResponse)

    const userData = await UserRegistrationService.signUpUser(signUpUserRequest)

    expect(axios.post).toHaveBeenCalledTimes(1)
    expect(userData).toEqual(userLoginResponse)
    expect(userData).toMatchSnapshot()
  })

  it('should return an object with error status and message if sign-up request is incorrect', async () => {
    axios.post.mockImplementationOnce(() => Promise.reject(errorSignUpResponse))

    await expect(UserRegistrationService.signUpUser(badSignUpUserRequest)).rejects.toMatchObject(
      errorSignUpResponse,
    )
    expect(badSignUpUserRequest).toMatchSnapshot()
  })

  it('should return an object with success status and message if log out user request success', async () => {
    axios.post = jest.fn().mockResolvedValue(logOutUserResponse)

    const logOut = await UserRegistrationService.logOutUser()

    expect(axios.post).toHaveBeenCalledTimes(1)
    expect(logOut).toEqual(logOutUserResponse)
    expect(logOut).toMatchSnapshot()
  })

  it('should return an object with error status and message if log out user request is incorrect', async () => {
    axios.post.mockImplementationOnce(() => Promise.reject(errorLogOutResponse))

    await expect(UserRegistrationService.logOutUser()).rejects.toMatchObject(errorLogOutResponse)
  })

  it('should return user data and new tokens after successful refresh request', async () => {
    axios.post = jest.fn().mockResolvedValue(userLoginResponse)

    const refresh = await UserRegistrationService.refresh()

    expect(axios.post).toHaveBeenCalledTimes(1)
    expect(refresh).toEqual(userLoginResponse)
    expect(refresh).toMatchSnapshot()
  })

  it('should return an object with error status and message if refresh request is incorrect', async () => {
    axios.post.mockImplementationOnce(() => Promise.reject(unauthorizedErrorResponse))

    await expect(UserRegistrationService.refresh()).rejects.toMatchObject(unauthorizedErrorResponse)
  })

  it('should return user data after getUserInfo request successfully', async () => {
    axios.get = jest.fn().mockResolvedValue(userLoginResponse)

    const userData = await UserRegistrationService.getUserInfo()

    expect(axios.get).toHaveBeenCalledTimes(1)
    expect(userData).toEqual(userLoginResponse)
    expect(userData).toMatchSnapshot()
  })

  it('should return an object with error status and message if getUserInfo request is incorrect', async () => {
    axios.get.mockImplementationOnce(() => Promise.reject(unauthorizedErrorResponse))

    await expect(UserRegistrationService.getUserInfo()).rejects.toMatchObject(
      unauthorizedErrorResponse,
    )
  })
})
