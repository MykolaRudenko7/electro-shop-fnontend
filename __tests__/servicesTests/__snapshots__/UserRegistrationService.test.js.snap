// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth service should return an object with error status and message if sign-in request is incorrect 1`] = `
{
  "email_signIn": undefined,
  "password_signIn": "wrongPassword",
}
`;

exports[`Auth service should return an object with error status and message if sign-up request is incorrect 1`] = `
{
  "confirmPassword_newUser": "wrongPassword",
  "email_newUser": "wrongEmail@gmail.com",
  "mobileNumber_newUser": "+wrongNumber",
  "name_newUser": undefined,
  "password_newUser": "wrongPassword",
}
`;

exports[`Auth service should return an object with success status and message if log out user request success 1`] = `
{
  "message": "Log out successfully",
  "success": true,
}
`;

exports[`Auth service should return user data after getUserInfo request successfully 1`] = `
{
  "data": [
    {
      "accessToken": undefined,
      "message": "Log in successfully",
      "refreshToken": undefined,
      "success": true,
      "user": {
        "__v": 0,
        "_id": undefined,
        "activationlink": undefined,
        "email": undefined,
        "isActivated": false,
        "lockUntil": null,
        "loginAttempts": 0,
        "mobileNumber": undefined,
        "name": undefined,
        "password": undefined,
      },
    },
  ],
  "status": 200,
}
`;

exports[`Auth service should return user data after successful sign in request 1`] = `
{
  "data": [
    {
      "accessToken": undefined,
      "message": "Log in successfully",
      "refreshToken": undefined,
      "success": true,
      "user": {
        "__v": 0,
        "_id": undefined,
        "activationlink": undefined,
        "email": undefined,
        "isActivated": false,
        "lockUntil": null,
        "loginAttempts": 0,
        "mobileNumber": undefined,
        "name": undefined,
        "password": undefined,
      },
    },
  ],
  "status": 200,
}
`;

exports[`Auth service should return user data after successful sign up request 1`] = `
{
  "data": [
    {
      "accessToken": undefined,
      "message": "Log in successfully",
      "refreshToken": undefined,
      "success": true,
      "user": {
        "__v": 0,
        "_id": undefined,
        "activationlink": undefined,
        "email": undefined,
        "isActivated": false,
        "lockUntil": null,
        "loginAttempts": 0,
        "mobileNumber": undefined,
        "name": undefined,
        "password": undefined,
      },
    },
  ],
  "status": 200,
}
`;

exports[`Auth service should return user data and new tokens after successful refresh request 1`] = `
{
  "data": [
    {
      "accessToken": undefined,
      "message": "Log in successfully",
      "refreshToken": undefined,
      "success": true,
      "user": {
        "__v": 0,
        "_id": undefined,
        "activationlink": undefined,
        "email": undefined,
        "isActivated": false,
        "lockUntil": null,
        "loginAttempts": 0,
        "mobileNumber": undefined,
        "name": undefined,
        "password": undefined,
      },
    },
  ],
  "status": 200,
}
`;

exports[`UserRegistrationService API requests should get user information successfully 1`] = `
{
  "data": [
    {
      "accessToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkIjoiNjU4MWEzMWI1ZDRiZjNiMjliMmQzZDMyIiwiYWNjZXNzIjp0cnVlLCJpYXQiOjE3MDM1OTgyODcsImV4cCI6MTcwMzU5OTE4N30.d1QyxH0Y-mxpglUBuyP0r7K6A_-LZCEvk1oqcsT5d6Y",
      "message": "Log in successfully",
      "refreshToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkIjoiNjU4MWEzMWI1ZDRiZjNiMjliMmQzZDMyIiwicmVmcmVzaCI6dHJ1ZSwiaWF0IjoxNzAzNTk4Mjg3LCJleHAiOjE3MDYxOTAyODd9.lf11OG_fQMX8NgUK9OCMTdeHZptVSgQN4xA4JqNTVfg",
      "success": true,
      "user": {
        "__v": 0,
        "_id": "6581a31b5d4bf3b29b2d3d32",
        "activationlink": "95709b2b-d686-47ae-8933-342250ce1509",
        "email": "m.rudenko.rb@gmail.com",
        "isActivated": false,
        "lockUntil": null,
        "loginAttempts": 0,
        "mobileNumber": "+380951970401",
        "name": "Mykola Rudenko",
        "password": "$2b$10$DMO2paWHwQiS24sw4t4ZUeCHTLiUR7V9pbqpfysTrazSakJgQjVSO",
      },
    },
  ],
  "status": 200,
}
`;

exports[`UserRegistrationService API requests should handle error when sign in data incorrect 1`] = `
{
  "email_signIn": "wrongEmail@gmail.com",
  "password_signIn": "wrongPassword",
}
`;

exports[`UserRegistrationService API requests should handle error when sign up data incorrect 1`] = `
{
  "confirmPassword_newUser": "wrongPassword",
  "email_newUser": "wrongEmail@gmail.com",
  "mobileNumber_newUser": "+wrongNumber",
  "name_newUser": "Bad Req",
  "password_newUser": "wrongPassword",
}
`;

exports[`UserRegistrationService API requests should log out user successfully 1`] = `
{
  "message": "Log out successfully",
  "success": true,
}
`;

exports[`UserRegistrationService API requests should sign in successfully 1`] = `
{
  "data": [
    {
      "accessToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkIjoiNjU4MWEzMWI1ZDRiZjNiMjliMmQzZDMyIiwiYWNjZXNzIjp0cnVlLCJpYXQiOjE3MDM1OTgyODcsImV4cCI6MTcwMzU5OTE4N30.d1QyxH0Y-mxpglUBuyP0r7K6A_-LZCEvk1oqcsT5d6Y",
      "message": "Log in successfully",
      "refreshToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkIjoiNjU4MWEzMWI1ZDRiZjNiMjliMmQzZDMyIiwicmVmcmVzaCI6dHJ1ZSwiaWF0IjoxNzAzNTk4Mjg3LCJleHAiOjE3MDYxOTAyODd9.lf11OG_fQMX8NgUK9OCMTdeHZptVSgQN4xA4JqNTVfg",
      "success": true,
      "user": {
        "__v": 0,
        "_id": "6581a31b5d4bf3b29b2d3d32",
        "activationlink": "95709b2b-d686-47ae-8933-342250ce1509",
        "email": "m.rudenko.rb@gmail.com",
        "isActivated": false,
        "lockUntil": null,
        "loginAttempts": 0,
        "mobileNumber": "+380951970401",
        "name": "Mykola Rudenko",
        "password": "$2b$10$DMO2paWHwQiS24sw4t4ZUeCHTLiUR7V9pbqpfysTrazSakJgQjVSO",
      },
    },
  ],
  "status": 200,
}
`;

exports[`UserRegistrationService API requests should sign up successfully 1`] = `
{
  "data": [
    {
      "accessToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkIjoiNjU4MWEzMWI1ZDRiZjNiMjliMmQzZDMyIiwiYWNjZXNzIjp0cnVlLCJpYXQiOjE3MDM1OTgyODcsImV4cCI6MTcwMzU5OTE4N30.d1QyxH0Y-mxpglUBuyP0r7K6A_-LZCEvk1oqcsT5d6Y",
      "message": "Log in successfully",
      "refreshToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkIjoiNjU4MWEzMWI1ZDRiZjNiMjliMmQzZDMyIiwicmVmcmVzaCI6dHJ1ZSwiaWF0IjoxNzAzNTk4Mjg3LCJleHAiOjE3MDYxOTAyODd9.lf11OG_fQMX8NgUK9OCMTdeHZptVSgQN4xA4JqNTVfg",
      "success": true,
      "user": {
        "__v": 0,
        "_id": "6581a31b5d4bf3b29b2d3d32",
        "activationlink": "95709b2b-d686-47ae-8933-342250ce1509",
        "email": "m.rudenko.rb@gmail.com",
        "isActivated": false,
        "lockUntil": null,
        "loginAttempts": 0,
        "mobileNumber": "+380951970401",
        "name": "Mykola Rudenko",
        "password": "$2b$10$DMO2paWHwQiS24sw4t4ZUeCHTLiUR7V9pbqpfysTrazSakJgQjVSO",
      },
    },
  ],
  "status": 200,
}
`;

exports[`UserRegistrationService API requests should update tokens successfully 1`] = `
{
  "data": [
    {
      "accessToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkIjoiNjU4MWEzMWI1ZDRiZjNiMjliMmQzZDMyIiwiYWNjZXNzIjp0cnVlLCJpYXQiOjE3MDM1OTgyODcsImV4cCI6MTcwMzU5OTE4N30.d1QyxH0Y-mxpglUBuyP0r7K6A_-LZCEvk1oqcsT5d6Y",
      "message": "Log in successfully",
      "refreshToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXlsb2FkIjoiNjU4MWEzMWI1ZDRiZjNiMjliMmQzZDMyIiwicmVmcmVzaCI6dHJ1ZSwiaWF0IjoxNzAzNTk4Mjg3LCJleHAiOjE3MDYxOTAyODd9.lf11OG_fQMX8NgUK9OCMTdeHZptVSgQN4xA4JqNTVfg",
      "success": true,
      "user": {
        "__v": 0,
        "_id": "6581a31b5d4bf3b29b2d3d32",
        "activationlink": "95709b2b-d686-47ae-8933-342250ce1509",
        "email": "m.rudenko.rb@gmail.com",
        "isActivated": false,
        "lockUntil": null,
        "loginAttempts": 0,
        "mobileNumber": "+380951970401",
        "name": "Mykola Rudenko",
        "password": "$2b$10$DMO2paWHwQiS24sw4t4ZUeCHTLiUR7V9pbqpfysTrazSakJgQjVSO",
      },
    },
  ],
  "status": 200,
}
`;
